### PR TITLE
Add testing and licensing pages, refresh publications and menu

### PR DESCRIPTION
**Pull request by timlichtenberg**
## What this does

Adds a new Testing page describing how the PROTEUS ecosystem is tested, refreshes the publications list with the latest papers, updates the Zalmoxis entry on the existing licensing page, and adds the Testing and License pages to the top menu.

## Why

The site had no public account of how the code is tested, and the publications list was missing recent work. The testing page explains the continuous-integration setup, public coverage tracking, the standard every test is held to, and how the framework is checked against telescope data and against independent codes.

## Changes

- New Testing page (`_pages/testing.md`, with data in `_data/test_counts.yml`): the continuous-integration pipelines, live and static test-count and coverage badges, the three categories of test, a required-and-rejected description of what makes a valid test, and worked examples of validation against JWST data and against the CHILI code intercomparison.
- Publications (`_pages/publications.md`): add a Preprints section for the submitted CHILI intercomparison paper, add the accepted study on oxygen-fugacity-dependent melting to the published list, and update the sub-Neptune solidification paper and the L 98-59 d paper to their final journal volumes and pages.
- Licensing page (`_pages/license.md`): move Zalmoxis to the Apache 2.0 row, matching its relicensing.
- Add Testing and License to the site menu, with License placed before People and Contact.

## Testing

Built the site locally with Jekyll and confirmed every changed page renders, the new sections and badges display, and the documentation and DOI links resolve. The test-count badges pull live from each repository's CI.

### CODE REVIEW

File: _pages/testing.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+  }
+
+  .footnote {
+    font-size: 0.82rem;
+    color: var(--muted-color);
+    margin-top: 0.4rem;
+  }
+</style>
+
+<p class="lead">Every change to the PROTEUS ecosystem clears an automated test gate before it can merge. Thousands of tests run across a dozen open-source repositories; line coverage is tracked publicly and allowed only to increase; and the core physics is cross-checked against independent codes and against refereed, published applications.</p>
```
Comment (nichollsh) [inline _pages/testing.md]: This is very "wordy" and also the specificity about a "dozen" will make it hard to maintain.

```suggestion
<p class="lead">Every change to code within the PROTEUS ecosystem must pass test 'gates' before it can be merged into the main versions of our models. In addition, the modelled physics is cross-checked against independent codes and against peer-reviewed papers. The 'coverage' of these tests (i.e. the fraction of the code which is probed) is tracked publicly. Validating workflows ensure that the coverage is only allowed only to increase as development progresses. We now have thousands of tests across all of our open-source repositories.</p>
```

File: _pages/testing.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+<div class="epistemic-callout">
+  <span class="callout-label">On what testing can and cannot show</span>
+  <p>"How do you know the code is right?" holds two questions. First: does the code do what we wrote it to do, every time? That is what the test suite proves, and this page shows how. Second: does what we wrote capture the planet? That is judged not by tests but by data: JWST spectra, geochemical and geophysical measurements, and comparison with independent codes.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
  <p>How do we know the code is accurately modelling the real world? This can be approached as two questions. Firstly: does the code do what we wrote it to do, every time? Our test suite tracks the modelled behaviours for reproducibility, and this page describes how. Secondly: do codes and models we have written capture the real behaviour of planets? This is judged by comparison against empirical data, including JWST spectra, geochemical and geophysical measurements, as well as with independent codes from other research groups.</p>
```

File: _pages/testing.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+  <p>The test suite is the precondition, not the substitute. It guarantees the code does, reproducibly, what we wrote, so that any disagreement with data is a statement about the physics we chose, not a bug we failed to catch.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
  <p>The test suite is a precondition for iterative model development, not the substitute for validation by humans. The tests guarantee that the code does what we intended reproducibly, so that any disagreement with data is a statement about the physics we chose, not a bug we failed to catch.</p>
```

File: _pages/testing.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+    </tr>
+    <tr>
+      <td><strong>Nightly full suite</strong></td>
+      <td>Nightly (03:00 UTC)</td>
+      <td>Runs the complete suite including integration and slow tests, uploads coverage to Codecov, and ratchets the coverage threshold so it can only ever increase.</td>
+    </tr>
+  </tbody>
+</table>
+
+Pull requests must additionally clear an 80% diff-cover bar on changed lines, and a 0.3% grace allowance on the global coverage threshold. The full developer-facing detail lives in the PROTEUS <a href="https://proteus-framework.org/PROTEUS/How-to/testing.html" target="_blank" rel="noopener noreferrer">testing how-to</a> and <a href="https://proteus-framework.org/PROTEUS/Explanations/test_framework.html" target="_blank" rel="noopener noreferrer">test-framework explanation</a>.
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
Pull requests must additionally clear an 80% bar on changed lines: meaning that 80% of the lines of code updated in a pull request must be tested. We apply a 0.3% grace tolerance on changes to the global coverage for each pull request. 

The full developer-facing detail is on the PROTEUS <a href="https://proteus-framework.org/PROTEUS/How-to/testing.html" target="_blank" rel="noopener noreferrer">testing how-to</a> and <a href="https://proteus-framework.org/PROTEUS/Explanations/test_framework.html" target="_blank" rel="noopener noreferrer">test-framework pages</a>.
```

File: _pages/testing.md [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+
+<h4>Three categories of test, three things they catch</h4>
+
+<div class="test-cards">
+  <div class="test-card">
+    <span class="accent accent-unit"></span>
+    <h5>Unit tests</h5>
+    <p>Isolate one function with mocked physics. Catch algebra errors, IO regressions, and broken assumptions about types and units. They run in milliseconds, so the suite stays cheap to invoke on every commit.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
    <p>Isolate one function with mocked physics. Catch algebra errors, input-output regressions, and broken assumptions about types and units. They run in milliseconds, so the suite stays cheap to invoke on every commit.</p>
```

File: _pages/testing.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+  </div>
+  <div class="test-card">
+    <span class="accent accent-integ"></span>
+    <h5>Integration and smoke tests</h5>
+    <p>Run real binaries on minimal configurations, often a single time step at low resolution. Catch coupling bugs that unit tests cannot see: when two correct modules disagree on a boundary condition, on a sign convention, or on which way time runs.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
    <p>Run the real models under a minimal-configuration scenario. Often, using a single simulation time step at spatial low resolution. These catch bugs that could arise due to model coupling, which specific unit tests cannot see. For example, when two correct modules disagree on a boundary condition, on a sign convention, or on which way time runs.</p>
```

File: _pages/testing.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+
+<h4>What counts as a test</h4>
+
+<p>Coverage measures how much of the code runs under test; it does not say whether a test would notice when the code is wrong. A test that passes for the wrong reason is worse than none, because it manufactures confidence. Every test in the suite is held to a standard a happy-path test cannot meet.</p>
+
+<div class="quality-cols">
+  <div class="quality-col quality-req">
+    <h5>Required in every test</h5>
+    <ul>
+      <li>A boundary or edge case: a temperature at the solidus, an empty inventory, or an extreme physical parameter.</li>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
      <li>A boundary or edge case: a temperature at the solidus, an empty inventory of volatiles, or an extreme physical parameter.</li>
```

File: _pages/testing.md [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+      <li>A value a plausible bug would break. A wrong sign, a wrong exponent, a missing factor, or a unit slip must make the test fail, so inputs are chosen where a wrong formula lands far from the right one.</li>
+      <li>On physics modules, at least one physical invariant: conservation of mass, energy, or angular momentum; boundedness (temperatures positive, mass fractions within zero and one); monotonicity or symmetry; or a value pinned to a published benchmark or an independent code.</li>
+    </ul>
+  </div>
+  <div class="quality-col quality-rej">
+    <h5>Rejected automatically</h5>
+    <ul>
+      <li>Single-assertion tests, and weak checks such as "the result is not empty" or "the result is positive" left to stand alone.</li>
+      <li>Floating-point equality against a literal value.</li>
+      <li>Tests that mock the very function they are meant to check.</li>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
      <li>Tests that used a dummy 'mocked' implementation of the very function they are meant to check.</li>
```

File: _pages/testing.md [hunk 9]
```diff
@@ -0,0 +1,376 @@
...
+      <li>Tests that never state which physical scenario or contract clause they verify.</li>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
      <li>Tests which never explicitly state their targeted physical scenario or 'contract clause'  (i.e. some obligation to reproduce a certain physical behaviour) </li>
```

File: _pages/testing.md [hunk 10]
```diff
@@ -0,0 +1,376 @@
...
+        {%- else -%}
+          <span style="color: var(--muted-color); font-size: 0.85rem;">No CI yet</span>
+        {%- endif -%}
+      </td>
+    </tr>
+    {% endfor %}
+  </tbody>
+</table>
+
+<p class="footnote">Rows displaying a numeric badge are live: the count is fetched from a JSON file that each repository's CI regenerates and republishes whenever its tests change. The remaining plain-number rows are static counts, last verified on 2026-07-04 by direct repository inspection. Ecosystem-wide adoption of the PROTEUS marker convention (unit, smoke, integration, slow) is in progress; until then, total counts are the most comparable figure across repositories.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
<p class="footnote">Rows displaying a numeric badge are live. The count is fetched from each repository's automated workflows, and republished whenever the repository's tests change. The remaining plain number rows are static counts ( last verified on 4 July 2026). Ecosystem-wide adoption of the PROTEUS marker convention (unit, smoke, integration, slow) is a point of active development.</p>
```

File: _pages/testing.md [hunk 11]
```diff
@@ -0,0 +1,376 @@
...
+
+<h4 id="tests-and-papers">Tests and papers</h4>
+
+Two tests of the framework sit beyond CI, and both are on the public record: whether a single code matches observations, and whether independent codes agree with one another. The papers highlighted below are part of the complete <a href="/publications">PROTEUS publications list</a>.
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
Testing extends far beyond software workflows. We also probe whether each code within the PROTEUS ecosystem matches real world observations, and whether independent codes (models) agree with one another in the expected regimes. These other testing modes are both on the public record. The papers highlighted below are part of the complete <a href="/publications">PROTEUS publications list</a>.
```

File: _pages/testing.md [hunk 12]
```diff
@@ -0,0 +1,376 @@
...
+  <p class="ref">Lichtenberg et al., <em>Planetary Science Journal</em> 7, 108 (2026). <a href="https://doi.org/10.3847/PSJ/ae593b">doi:10.3847/PSJ/ae593b</a></p>
+</div>
+
+<div class="vignette">
+  <h6>CHILI I: primordial magma oceans of Earth and Venus</h6>
+  <p>The first results benchmark planetary evolution codes on the primordial magma oceans of Earth and Venus. Nominal Earth models agree on solidification timescale to within 4 Myr; Venus scenarios diverge more, and the intercomparison pins that divergence to specific treatments of volatile partitioning, vertical energy transport, mantle rheology, and melting. Locating those sensitivities is what separates a code that runs from one that agrees with independent implementations, and shows exactly where it does not.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
  <p>The model intercomparison results benchmark planetary evolution codes against the scenarios of Earth and Venus. Earth models agree on solidification timescale to within 4 Myr. Simulated Venus scenarios diverge more, between the participating models. The intercomparison highlights that divergence arise through variant treatments of volatile partitioning, vertical energy transport, mantle rheology, and melting. Locating those sensitivities is what separates a code that runs from one that agrees with independent implementations.</p>
```

File: _pages/testing.md [hunk 13]
```diff
@@ -0,0 +1,376 @@
...
+</div>
+
+<h5 class="subsection">Tested against independent codes: the CHILI intercomparison</h5>
+
+<p class="subsection-lead">Publication tests one code against data. A harder test is whether independent codes, built by different groups with different numerical choices, agree on the same problem. We initiated the Coupled atmospHere Interior modeL Intercomparison (CHILI), the first shared benchmark suite for coupled interior-atmosphere models, to make that test rigorous.</p>
+
+<div class="vignette">
+  <h6>CHILI: a shared benchmark for coupled codes</h6>
+  <p>CHILI defines a common protocol of standardized cases, early Earth, Venus, and rocky exoplanets around M-dwarfs, that any coupled interior-atmosphere code can run. Fixing the inputs turns model disagreement into a measured quantity: differences in energy transport, geochemistry, and volatile cycling become numbers to explain rather than buried assumptions. It is the community-scale counterpart of the cross-code checks PROTEUS runs internally, such as SPIDER against Aragog for the magma-ocean energy budget.</p>
```
Comment (nichollsh) [inline _pages/testing.md]:
```suggestion
  <p><a href="https://nexss.info/cuisines/chili-mip/">CHILI defines a common protocol of standardized cases</a>, early Earth, Venus, and rocky exoplanets around M-dwarfs, that any coupled interior-atmosphere code can run. Fixing the inputs turns model disagreement into a measured quantity: differences in energy transport, geochemistry, and volatile cycling become numbers to explain rather than buried assumptions. It is the community-scale counterpart of the cross-code checks PROTEUS runs internally, such as SPIDER against Aragog for the magma-ocean energy budget.</p>
```

**Review by nichollsh — CHANGES_REQUESTED** [review]
Thanks for this, @timlichtenberg. These are some important additions which highlight the robustness of our testing and approach.

I feel that there is a lot of jargon in the current text. It's also quite wordy and hard to parse.

I've made suggestions below which eliminate some abbreviations, add clarifications for jargon or avoid jargon entirely, and improve the readability of the prose.

**Comment by timlichtenberg** [comment]
I took all your changes, very good suggestions, thansk! :)

**Review by nichollsh — DISMISSED** [review]
